## Problem #301: Nim
[Link to Problem](https://projecteuler.net/problem=301)

### Approach

We can let the solution bruteforce all the numbers and it should take ~4 minutes or we can notice that for increasing powers of two the number of valid values for $n$ is the $(n+2)^{th}$ element of the Fibonacci sequence.

In [23]:
exponent = 30

# Bruteforce:
# limit = 1 << exponent

# count = 0

# for x in range(1, limit + 1):
#     if  x ^ (2 * x) ^ (3 * x) == 0:
#         count += 1

# print(count)

fibonacci = [1, 1]
for _ in range(exponent):
    fibonacci.append(fibonacci[-1] + fibonacci[-2])
print(fibonacci[-1])

2178309


###### Result: **2178309** | Execution time: 0s

### Complexity analysis

Time & space complexity: $O(n)$

##### Tags: #fibonacci, #bruteforce